### PR TITLE
Adds a join indicator

### LINKED ISSUE #233: 'indicator' option for DataFrame#join method.
I request 'indicator' option for DataFrame#join method. Like [pandas's one](http://pandas.pydata.org/pandas-docs/stable/merging.html#the-merge-indicator).

### PR DESCRIPTION
**Pull request by gnilrets**
Resolves #233

### CODE REVIEW

File: spec/core/merge_spec.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@
       expect(@left.join(@right_many, how: :inner, on: [:id])).to eq(answer)
     end
 
-    it "performs a full outer join" do
+    it "performs a full outer join", wip2: true do
```
Comment (v0dro) [inline spec/core/merge_spec.rb]: `wip2: true` as in?

**Comment by v0dro** [comment]
Could you add information about the new option in the documentation?

**Comment by gnilrets** [comment]
Thanks @v0dro.  Changes are in the latest commit.

**Comment by v0dro** [comment]
Failing tests :P

**Comment by gnilrets** [comment]
Ok, let's try that again @v0dro.  Thanks!

**Pull request merged**